src/routing_table/network_tests.rs

```diff
@@ -242,48 +242,45 @@ fn node_to_group_message() {
     }
 }
 
-fn verify_invariant(network: &mut Network) {
-    let keys = network.keys();
-    for _ in 0..20 {
-        let address = network.rng.gen();
-        let close_peer = network.close_node(address);
-        let group = unwrap!(network.nodes[&close_peer].close_names(&address));
-        for &node in &keys {
-            match network.nodes[&node].close_names(&address) {
-                None => assert!(!group.contains(&address)),
-                Some(nodes) => {
-                    if network.nodes[&node].is_recipient(&Destination::Group(address)) {
-                        assert_eq!(group, nodes);
-                    } else {
-                        for candidate in &group {
-                            assert!(nodes.contains(&candidate));
-                        }
-                    }
-                }
+fn verify_invariant(network: &Network) {
+    let mut groups: HashMap<Prefix<u64>, HashSet<u64>> = HashMap::new();
+    // first, collect all groups in the network
+    for node in network.nodes.values() {
+        for prefix in node.groups.keys() {
+            let mut group_content = node.groups[prefix].clone();
+            if *prefix == node.our_group_prefix {
+                group_content.insert(*node.our_name());
             }
+            if let Some(group) = groups.get_mut(prefix) {
+                group.extend(group_content);
```

Comment:
Let's assert that `group` equals `groups.get_mut(prefix)`. If that fails, that points to a bug, too (although possibly one in the node adding handling in the test).